testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5185,6 +5187,61 @@ void testBitGetFunc(SqlOperatorFixture f, String functionName) {
     }
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6551">[CALCITE-6551]
+   * Add DATE_FORMAT function (enabled in MySQL library)</a>. */
+  @Test void testDateFormat() {
+    final SqlOperatorFixture f = fixture()
+        .withTester(t -> TESTER).withLibrary(SqlLibrary.MYSQL);
+    f.setFor(SqlLibraryOperators.DATE_FORMAT);
+    final Locale originalLocale = Locale.getDefault();
+
+    try {
+      Locale.setDefault(Locale.US);
+      f.checkString("date_format(timestamp '2009-10-04 22:23:00', '%W %M %Y')",
```

Comment:
Does 0000-01-01 or BC date support format?

Reply:
I tested this example in mysql.
`SELECT DATE_FORMAT('0000-01-01', '%X %V');`
return 
0000-01
but
`SELECT DATE_FORMAT(timestamp '0000-01-01', '%X %V');`
return error: Incorrect DATETIME value: '0000-01-01'

Reply:
> It throws an exception:Illegal TIMESTAMP literal, In the timestamp scenario, this is consistent with mysql's result.

In the Calcite test, this throws an exception because it can't be converted to a timestamp

Reply:
Yes, Calcite has its own date standard.

Reply:
> SELECT DATE_FORMAT(timestamp '0000-01-01', '%X %V');

This failed because '0000-01-01' is not a timestamp. You can try:
```
SELECT DATE_FORMAT(data '0000-01-01', '%X %V');
```

Reply:
> > SELECT DATE_FORMAT(timestamp '0000-01-01', '%X %V');
> 
> This failed because '0000-01-01' is not a timestamp. You can try:
> 
> ```
> SELECT DATE_FORMAT(data '0000-01-01', '%X %V');
> ```

0000-01-01 is not even a valid date according to the SQL standard.